pkg/util/test/util.go

```diff
@@ -1496,11 +1496,16 @@ func GetExpectedPxNodeList(cluster *corev1.StorageCluster) ([]v1.Node, error) {
 			NodeAffinity: cluster.Spec.Placement.NodeAffinity.DeepCopy(),
 		}
 	} else {
+		runOnMaster := false
+		if IsK3sCluster() || IsPxDeployedOnMaster(cluster) {
+			runOnMaster = true
+		}
+
 		dummyPod.Spec.Affinity = &v1.Affinity{
-			NodeAffinity: defaultPxNodeAffinityRules(IsK3sCluster()),
+			NodeAffinity: defaultPxNodeAffinityRules(runOnMaster),
 		}
 
-		if IsK3sCluster() {
+		if IsK3sCluster() || IsPxDeployedOnMaster(cluster) {
```

Comment:
Why not use the runOnMaster local variable here?

Reply:
Good point!

Reply:
Fixed